pkg/systemd/generate/containers_test.go

```diff
@@ -546,7 +627,15 @@ Environment=PODMAN_SYSTEMD_UNIT=%n
 Restart=on-failure
 TimeoutStopSec=70
 ExecStartPre=/bin/rm -f %t/%n.ctr-id
-ExecStart=/usr/bin/podman run --cidfile=%t/%n.ctr-id --cgroups=no-conmon --rm --sdnotify=conmon -d --replace --name test awesome-image:latest sh -c "kill $$$$ && echo %%\\"
+ExecStart=/usr/bin/podman run \
+	--cidfile=%t/%n.ctr-id \
+	--cgroups=no-conmon \
+	--rm \
+	--sdnotify=conmon \
+	-d \
+	--replace \
+	--name test awesome-image:latest sh \
+	-c "kill $$$$ && echo %%\\"
```

Comment:
```suggestion
	sh -c "kill $$$$ && echo %%\\"
```
and remove "sh" from line 637

Reply:
That is difficult/impossible to do, would need to figure out the first line of the command. Recall these are commands within commands.  The command is not processed by Podman so there is no way to differentiate between the command/image and subcommand.

Reply:
It is possible to do this but it would require way more logic. I think the current check is good enough.

Reply:
While possible for certain use cases, like podman in podman, I am not sure we would be 100 successful in interpreting these type of commands imagine running chroot or sudo within a container, or even bash -c.  It would get way more complex for not much benefit.